boards/arm/v2m_musca/doc/v2m_musca.rst

```diff
@@ -0,0 +1,412 @@
+
+QSPI boot memory
+================
+Normal Musca-A test chip boot operation is from external QSPI 8MB flash memory.
+Only the lowest 256KB of QSPI memory is directly accessible.
+More memory is accessible through indirect addressing.
+
+Programming and Debugging
+*************************
+
+Musca supports the v8m security extension, and by default boots to the secure
+state.
+
+When building a secure(S)/non-secure(NS application, the S one will
+have to set the idau/sau and mpc configuration to permit access from the NS,
+before jumping.
+
+The following system components are required to be properly configured during the
+secure firmware:
+
+- AHB5 TrustZone Memory Protection Controller (MPC).
+- AHB5 TrustZone Peripheral Protection Controller (PPC).
+- Implementation-Defined Attribution Unit (IDAU).
+
+For more details please refer to `Corelink SSE-200 Subsystem`_.
+
+Flashing
+========
+
+DAPLink
+---------
+
+V2M Musca provides:
+
+- A USB connection to the host computer, which exposes a Mass Storage and an
+  USB Serial Port.
+- A Serial Flash device, which implements the USB flash disk file storage.
+- A physical UART connection which is relayed over interface USB Serial port.
+
+This interfaces are exposed via DAPLink which provides:
+
+- Serial Wire Debug (SWD).
+- USB Mass Storage Device (USBMSD).
+- UART.
+- Remote reset.
+
+For more details please refer
+to `DAPLink Website`_.
+
+
+Building a secure only application
+----------------------------------
+
+
+You can build applications in the usual way. Here is an example for
+the :ref:`hello_world` application.
+
+
+Building a secure/non-secure with Zephyr on both sides
+------------------------------------------------------
+
+The process is similar to before, with the added configuration of setting-up
+the secure/non-secure configuration entry on zephyr.
+
+
+Building a secure/non-secure with Trusted Firmware
+--------------------------------------------------
+
+The process requires five steps:
+
+1. Build Trusted Firmware (tfm).
+2. Import it as a library to the Zepyhr source folder.
+3. Build Zephyr with a non-secure configuration.
+4. Merge the two binaries together and sign them.
+5. Concatenate the bootloader with the signed image blob.
+
+In order to build tfm please refer to `Trusted Firmware M Guide`_.
+Follow the build steps for AN521 target while replacing the platform with
+``-DTARGET_PLATFORM=MUSCA_A`` and compiler (if required) with ``-DCOMPILER=GCC``
+
+Copy over tf-m as a library to the zephyr project source and create a shortcut
+for the secure veneers.
+
+.. code-block:: bash
+
+   cp -r install/ $ZEPHYR_PROJECT/src/ext
+   cp $ZEPHYR_PROJECT/src/ext/install/export/tfm/veneers/s_veneers.o $ZEPHYR_PROJECT/src/ext
+
+
+Build the Zephyr app in the usual way.
+
+Uploading an application to V2M Musca
+-------------------------------------
+
+In order to flash an application to V2M Musca, it needs to be converted to
```

Comment:
For clarity:

    Applications must be converted to Intel's hex format before
    being flashed to a VM2 Musca board.